src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -1896,6 +1901,66 @@ public static bool IsOfTypePreventDeletions(ReplicationBatchItem item)
 
             return false;
         }
+
+        internal ReplicationProcessProgress GetOutgoingReplicationProgress(DocumentsOperationContext documentsContext, DatabaseOutgoingReplicationHandler handler)
+        {
+            var lastProcessedEtag = handler.LastSentDocumentEtag;
+
+            var progress = new ReplicationProcessProgress
+            {
+                FromToString = handler.FromToString,
+                LastEtagSent = lastProcessedEtag,
+                DestinationChangeVector = handler.LastAcceptedChangeVector,
+                SourceChangeVector = handler.LastSentChangeVector,
+                AverageProcessedPerSecond = handler.Metrics.GetProcessedPerSecondRate() ?? 0.0
+            };
+
+            var collections = Database.DocumentsStorage.GetCollections(documentsContext).Select(x => x.Name).ToList();
```

Comment:
can we drop the `ToList()`?

```suggestion
            var collections = Database.DocumentsStorage.GetCollections(documentsContext).Select(x => x.Name);
```

Reply:
definitely! done